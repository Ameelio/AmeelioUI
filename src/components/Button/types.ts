export enum ButtonSize {
    SMALL = 'sm',
    LARGE = 'lg'
}

export enum ButtonType {
    BUTTON = 'button',
    SUBMIT = 'submit',
    RESET = 'reset',
}

export enum ButtonVariant {
    PRIMARY = 'primary',
    SECONDARY = 'secondary',
    SUCCESS = 'success',
    DANGER = 'danger',
    WARNING = 'warning',
    INFO = 'info',
    DARK = 'dark',
    LIGHT = 'light',
    LINK = 'link',
    OUTLINE_PRIMARY = 'outline-primary',
    OUTLINE_SECONDARY = 'outline-secondary',
    OUTLINE_SUCCESS = 'outline-success',
    OUTLINE_DANGER = 'outline-danger',
    OUTLINE_WARNING = 'outline-warning',
    OUTLINE_INFO = 'outline-info',
    OUTLINE_DARK = 'outline-dark',
    OUTLINE_LIGHT = 'outline-light',
}