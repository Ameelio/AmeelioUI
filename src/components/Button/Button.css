.y-button.btn.btn-secondary {
    color: white;
}