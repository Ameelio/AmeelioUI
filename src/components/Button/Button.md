### Examples

Basic button:

```js
<Button>Push Me</Button>
```

Secondary button:

```js
<Button
  variant='secondary'
>Push Me</Button>
```

Disabled button:

```js
<Button
  disabled={true}
>Push Me</Button>
```

Block button:

```js
<Button
  block={true}
>Push Me</Button>
```

Small button:

```js
<Button
  size='sm'
>Push Me</Button>
```

Large button:

```js
<Button
  size='lg'
>Push Me</Button>
```
