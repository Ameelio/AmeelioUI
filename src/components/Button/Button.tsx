import React, { Component } from 'react';
import BootstrapButton, { ButtonProps as BootstrapButtonProps } from 'react-bootstrap/Button';
import { ElementType } from 'react';

export interface ButtonProps extends BootstrapButtonProps {
    /**
     * Manually set the visual state of the button to :active
     * @default false
     */
    active?: boolean;

    /**
     * Custom element type for this component
     */
    as?: ElementType;

    /**
     * Span full width of the Button parent
     */
    block?: boolean;

    /**
     * Disables the Button, preventing mouse events
     * @default false
     */
    disabled?: boolean;

    /**
     * Providing href will render an <a> element, styled as a button
     */
    href?: string;

    /**
     * Specifies large or small button
     */
    size?: 'sm' | 'lg';

    /**
     * Defines HTML button type attribute
     * @default 'button''
     */
    type?: 'button' | 'reset' | 'submit';

    /**
     * One or more button variant combinations including outline variants
     * @default 'primary'
     */
    variant?:
        | 'primary'
        | 'secondary'
        | 'success'
        | 'danger'
        | 'warning'
        | 'info'
        | 'dark'
        | 'light'
        | 'link'
        | 'outline-primary'
        | 'outline-secondary'
        | 'outline-success'
        | 'outline-danger'
        | 'outline-warning'
        | 'outline-info'
        | 'outline-dark'
        | 'outline-light';

    /**
     * Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css
     * @default 'btn'
     */
    bsPrefix?: string;
}

export default class Button extends Component<ButtonProps> {
    public render() {
        return (
            <BootstrapButton {...this.props} />
        );
    }
}
