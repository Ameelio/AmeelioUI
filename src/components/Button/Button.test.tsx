import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Button, { ButtonProps, ButtonSize, ButtonType, ButtonVariant } from './Button';

const sampleText = 'Click Me';
const sampleUrl = 'ameelio.org';

describe('<Button />', () => {
    let component: ShallowWrapper<ButtonProps>;

    describe('with default options', () => {
        beforeEach(() => {
            component = shallow(<Button>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with block style', () => {
        beforeEach(() => {
            component = shallow(<Button block={true}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with disabled state', () => {
        beforeEach(() => {
            component = shallow(<Button disabled={true}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with link', () => {
        beforeEach(() => {
            component = shallow(<Button href={sampleUrl}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with link', () => {
        beforeEach(() => {
            component = shallow(<Button href={sampleUrl}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with small size', () => {
        beforeEach(() => {
            component = shallow(<Button size={ButtonSize.SMALL}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with large size', () => {
        beforeEach(() => {
            component = shallow(<Button size={ButtonSize.LARGE}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with submit type', () => {
        beforeEach(() => {
            component = shallow(<Button type={ButtonType.SUBMIT}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with reset type', () => {
        beforeEach(() => {
            component = shallow(<Button type={ButtonType.RESET}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with submit type', () => {
        beforeEach(() => {
            component = shallow(<Button type={ButtonType.SUBMIT}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with secondary variant', () => {
        beforeEach(() => {
            component = shallow(<Button variant={ButtonVariant.SECONDARY}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with outline warning variant', () => {
        beforeEach(() => {
            component = shallow(<Button variant={ButtonVariant.OUTLINE_WARNING}>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with custom base class prefix', () => {
        beforeEach(() => {
            component = shallow(<Button bsPrefix='ameelio-btn'>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with additional className', () => {
        beforeEach(() => {
            component = shallow(<Button className='TEST_CLASSNAME'>{sampleText}</Button>);
        });

        it('matches its snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });
});