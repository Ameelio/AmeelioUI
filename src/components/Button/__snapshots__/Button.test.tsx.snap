// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> with additional className matches its snapshot 1`] = `
<Button
  active={false}
  className="TEST_CLASSNAME"
  disabled={false}
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with block style matches its snapshot 1`] = `
<Button
  active={false}
  block={true}
  disabled={false}
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with custom base class prefix matches its snapshot 1`] = `
<Button
  active={false}
  bsPrefix="ameelio-btn"
  disabled={false}
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with default options matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with disabled state matches its snapshot 1`] = `
<Button
  active={false}
  disabled={true}
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with large size matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  size="lg"
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with link matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  href="ameelio.org"
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with link matches its snapshot 2`] = `
<Button
  active={false}
  disabled={false}
  href="ameelio.org"
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with outline warning variant matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  type="button"
  variant="outline-warning"
>
  Click Me
</Button>
`;

exports[`<Button /> with reset type matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  type="reset"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with secondary variant matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  type="button"
  variant="secondary"
>
  Click Me
</Button>
`;

exports[`<Button /> with small size matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  size="sm"
  type="button"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with submit type matches its snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  type="submit"
  variant="primary"
>
  Click Me
</Button>
`;

exports[`<Button /> with submit type matches its snapshot 2`] = `
<Button
  active={false}
  disabled={false}
  type="submit"
  variant="primary"
>
  Click Me
</Button>
`;
